### PR TITLE
fix(source/nats): add missing `default` attribute for vectors

### PR DESCRIPTION
**Pull request by benjamin-awd**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
Currently, the `backoff` and `filter_subjects` fields are required due to a missing `default` attribute, causing source creation to fail when they are not defined.

However, these fields should are supposed to be optional and should allow the creation of the source to proceed, even if they are not defined. 

This PR adds the missing `default` attribute, which fixes the above issue.

Related to  https://github.com/risingwavelabs/risingwave/pull/17615

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/connector/with_options_source.yaml [hunk 1]
```diff
@@ -537,6 +537,7 @@ NatsProperties:
   - name: consumer.filter_subjects
     field_type: Vec<String>
     required: false
+    default: Default::default
```
Comment (xiangjinwu) [inline src/connector/with_options_source.yaml:540]: Does this PR imply the following problems:
* Without the `serde(default)` the `required: false` here is actually misleading
* Other attributes without `serde(default)` have the same problem

Comment (benjamin-awd) [reply src/connector/with_options_source.yaml:540]: Based on our testing of this, the other attributes are unaffected. It seems that only the vector types are affected due - my guess is that this is somehow needed due to the custom deserializer.

I drew reference from the Iceberg sink that uses a similar pattern https://github.com/risingwavelabs/risingwave/blob/63b4d47ea9f54f567983e097e15d8dc9b0384849/src/connector/src/sink/iceberg/mod.rs#L129-L141

**Review by xiangjinwu — APPROVED** [review]

File: src/connector/with_options_source.yaml (hunk 1 above)
Comment (xiangjinwu) [reply src/connector/with_options_source.yaml:540]: Thanks! Yes, it seems only required when using `deserialize_with`. We will see if we can fix the `required: false` in yaml.

Comment (benjamin-awd) [reply src/connector/with_options_source.yaml:540]: Anyway just to confirm, these attributes are optional so I think it's fine to keep `required: false` :)

Thanks for taking a look!

Comment (xiangjinwu) [reply src/connector/with_options_source.yaml:540]: Yes these 2 are now truly optional with `default`. I meant fixing the yaml generator when `default` is missing but `deserialize_with` is present.

**Pull request merged**